src/main.rs

```diff
@@ -18,25 +18,119 @@ fn main() {
 }
 
 fn real_main() -> Result<i32, Box<dyn std::error::Error>> {
+    let rng = Rng::new();
     let cargo = std::env::var("CARGO").unwrap_or_else(|_| "cargo".to_owned());
     let mut arg_iter = std::env::args().peekable();
-    let _cargo = arg_iter.next();
-    if arg_iter.peek().map_or(false, |arg| arg == "mommy") {
-        let _mommy = arg_iter.next();
+
+    // Understand who mommy really is~
+    //
+    // *INHALES*
+    //
+    // Ok so first argument is this binary, should look like /some/path/to/cargo-mommy(.exe)
+    // but we want to let it instead be cargo-daddy, and for everything to rekey itself to:
+    // * make the default role be "daddy"
+    // * make all the read env-vars be "CARGO_DADDYS_*"
+    let mut platform_uses_exe = false;
+    let raw_first_arg = arg_iter.next().unwrap_or_default();
+    let first_arg = if let Some(stripped) = raw_first_arg.strip_suffix(".exe") {
```

Comment:
I don't love how this is done; I would prefer something along the lines of creating a `Path`, since this is fundamentally a path-operation, and then remove the extension. What happens if someone writes `CARGO-MOMMY.EXE`, for example? in that case, this would use `MOMMY.EXE` as the "key".

Reply:
Additionally, I don't understand why this is checking specifically for `.exe`; afaik, only on Windows is `cargo-blah.exe` the default for Rust (I know other programming languages still use `.exe`, but we're discussing Rust specifically).

Reply:
```suggestion
    let first_arg = if let Some(stripped) = raw_first_arg.strip_suffix(std::env::consts::EXE_SUFFIX) {
```